### ISSUE TITLE
`BitMap.resize` cause almost endless error spam

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#70187]
### Godot version

3.6.beta.custom_build.d79699041

### System information

Ubuntu 22.10 - AMD RX 470, Gnome shell 43 X11

### Issue description

```
extends Node
func _process(delta):

	var temp_variable2470 = BitMap.new()
	temp_variable2470.resize(Vector2(4040.38916, 9614.855469))
	temp_variable2470.resize(Vector2(INF, 6091.175293))
```
prints thousand/million times
```
ERROR: Index x = 33 is out of bounds (width = -2147483648).
   at: set_bit (scene/resources/bit_map.cpp:113)
ERROR: Index x = 33 is out of bounds (width = -2147483648).
   at: set_bit (scene/resources/bit_map.cpp:113)
ERROR: Index x = 33 is out of bounds (width = -2147483648).
   at: set_bit (scene/resources/bit_map.cpp:113)
ERROR: Index x = 33 is out of bounds (width = -2147483648).
   at: set_bit (scene/resources/bit_map.cpp:113)
ERROR: Index x = 33 is out of bounds (width = -2147483648).
   at: set_bit (scene/resources/bit_map.cpp:113)
ERROR: Index x = 34 is out of bounds (width = -2147483648).
   at: set_bit (scene/resources/bit_map.cpp:113)
ERROR: Index x = 34 is out of bounds (width = -2147483648).
   at: set_bit (scene/resources/bit_map.cpp:113)
ERROR: Index x = 34 is out of bounds (width = -2147483648).
```

### Steps to reproduce

Above

### Minimal reproduction project

N/A

**Comment by pfertyk** [comment]
I'd like to submit my fix ;) This is caused by `real` vs `int` in the `BitMap.resize` method. The conversion resulted (on my machine) in the following values:
* `width`: 4040
* `new_bitmap->width`: -2147483648
* `p_new_size.width`: -9223372036854775808

I think this is already fixed in `master` by using `Size2i` instead of `Size2`. As a quick fix, I've used `new_bitmap-width` instead of `p_new_size.width`.

**Comment by akien-mga** [comment]
Fixed by #70368.
